generic_scripts/nlk.py

```diff
@@ -0,0 +1,167 @@
+import numpy as _np
+import matplotlib.pyplot as _plt
+from mathphys.constants import vacuum_permeability
+from mathphys.beam_optics import beam_rigidity
+
+
+class Wire:
+    """."""
+    def __init__(self, position, current):
+        self.pos = position  # Column vector
+        self.curr = current
+
+    def Bfield(self, x, y):
+        mu0 = vacuum_permeability
+        r_w = self.pos  # Wire positions
+        r = _np.array([x, y])[:, None]  # Target position
+        rc = r - r_w  # Cursive r
+        theta = _np.arctan2(rc[1], rc[0])[0]
+        theta_vec = _np.array([-_np.sin(theta), _np.cos(theta)])[:, None]
+        B = mu0 * self.curr/(2*_np.pi*_np.linalg.norm(rc))*theta_vec + r_w
+        return B
+
+    def include_pos_error(self, sigma):
+        self.error = _np.random.normal(loc=0, scale=sigma)
+        self.pos = self.pos + self.error
+
+    @property
+    def pos(self):
+        return self._pos
+
+    @pos.setter
+    def pos(self, position):
+        self._pos = position
+
+    @property
+    def curr(self):
+        return self._curr
+
+    @curr.setter
+    def curr(self, current):
+        self._curr = current
+
+
+class NLK:
+    """."""
+    def __init__(self, curr=1850, errors=False, sigma_errors=1e-5):
+        # Setting wire positions
+        wire_positions = []
+        wire_positions.append(_np.array([7.0,  5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([10.,  5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([-7.,  5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([-10,  5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([7.0, -5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([10., -5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([-7., -5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([-10, -5.85])[:, None]*1e-3)
+        self._positions = wire_positions
+
+        # Setting wire currents
+        currents = []
+        for i in range(8):
+            if i % 2:
+                currents.append(-curr)
+            else:
+                currents.append(curr)
+        self._currents = currents
+
+        # Creating wires
+        self._wires = []
+        for i in range(8):
+            pos = wire_positions[i]
+            curr = currents[i]
+            wire = Wire(pos, curr)
+            if errors:
+                wire.include_pos_error(sigma=sigma_errors)
+            self._wires.append(wire)
+
+    @property
+    def wires(self):
+        return self._wires
+
+    @wires.setter
+    def wires(self, wires_objects):
+        self._wires = wires_objects
+
+    @property
+    def positions(self):
+        "Return positions of all wires"
+        return self._positions
+
+    @positions.setter
+    def positions(self, wires_positions):
```

Comment:
this setter method should in fact set the positions of each wire.

Reply:
Bug solved in [5da7e34](https://github.com/lnls-fac/apsuite/pull/178/commits/5da7e34230705947c7f1394521605248ec723351). But isn't possible yet to set the wire positions individually.